core/app/[locale]/(default)/product/[slug]/page.tsx

```diff
@@ -332,6 +332,7 @@ export default async function Product({ params: { locale, slug }, searchParams }
           src: product.defaultImage?.url ?? '',
           alt: product.defaultImage?.altText ?? '',
         }}
+        product={formattedProduct}
```

Comment:
Passing product so we can properly scope the snapshot/label the changes for the Publish dialog.